### ISSUE TITLE
docker: Error response from daemon: OCI runtime create failed:

### ISSUE THREAD

**Issue by hongyi-zhao** [Hagb/docker-easyconnect#17]
Today, when I try to run the Hagb/docker-easyconnect image, it failed with the followng error:

```
$ xhost +LOCAL:
non-network local connections being added to access control list
$ docker run --device /dev/net/tun --cap-add NET_ADMIN -ti -v /tmp/.X11-unix:/tmp/.X11-unix -v $HOME/.Xauthority:/root/.Xauthority -e EXIT=1 -e DISPLAY=$DISPLAY -e URLWIN=1 -e TYPE=x11 -v $HOME/.ecdata:/root -p 127.0.0.1:1080:1080 hagb/docker-easyconnect
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/home/werner/.Xauthority\\\" to rootfs \\\"/var/lib/docker/overlay2/972dfca71643bda2f2af2f93930092b43b5b1fc6e8af553af3a37fc1bd69420d/merged\\\" at \\\"/var/lib/docker/overlay2/972dfca71643bda2f2af2f93930092b43b5b1fc6e8af553af3a37fc1bd69420d/merged/root/.Xauthority\\\" caused \\\"not a directory\\\"\"": unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.
ERRO[0000] error waiting for container: context canceled 
non-network local connections being removed from access control list
werner@X10DAi:~/Public/repo/github.com/hongyi-zhao$ sudo systemctl restart docker
werner@X10DAi:~/Public/repo/github.com/hongyi-zhao$ bash docker-easyconnect.git.sh 
non-network local connections being added to access control list
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/home/werner/.Xauthority\\\" to rootfs \\\"/var/lib/docker/overlay2/b1115099b8eee370b3d4247904378684b1a5118f85d2503d28492e6a3c601494/merged\\\" at \\\"/var/lib/docker/overlay2/b1115099b8eee370b3d4247904378684b1a5118f85d2503d28492e6a3c601494/merged/root/.Xauthority\\\" caused \\\"not a directory\\\"\"": unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.
ERRO[0000] error waiting for container: context canceled 

```

Any hints for this problem?

Regards,
HY

**Comment by Hagb** [comment]
> `mounting \\\"/home/werner/.Xauthority\\\" to rootfs …/merged/root/.Xauthority\\\" caused \\\"not a directory\\\"\""`

Is there anything in `$HOME/.ecdata/.Xauthority`?

**Comment by hongyi-zhao** [comment]
```
werner@X10DAi:~$ sudo cat $HOME/.ecdata/.Xauthority
�1MIT-MAGIC-COOKIE-1ߋ�bz�+J�Qp2�i�
werner@X10DAi:~$ cat $HOME/.Xauthority
cat: /home/werner/.Xauthority: Is a directory
werner@X10DAi:~$ ls $HOME/.Xauthority
werner@X10DAi:~$ ls -la $HOME/.Xauthority
total 8
drwxr-xr-x  2 root   root   4096 Nov 12 22:29 .
drwxr-xr-x 51 werner werner 4096 Nov 15 14:48 ..
werner@X10DAi:~$ 
```

**Comment by Hagb** [comment]
Try to `sudo rm -r ~/.ecdata/.Xauthority`?
Did you run EasyConnect container in vnc with the same `ecdata` dir before the error happened?

**Comment by Hagb** [comment]
> I deleted the above files/folder, but still fail to log in. See the screenshot below:
> 
> ![image](https://user-images.githubusercontent.com/11155854/99179112-51064180-2755-11eb-9b10-9c53c1179f1a.png)

It seems not the same problem with the first one.

**Comment by hongyi-zhao** [comment]
Anyway, what's the possible reason for this problem?

**Comment by Hagb** [comment]
The first problem: (I guess) `.Xauthority` of vnc server is a regular file saved in `ecdata`, but `.Xauthority` of your x11 server is a directory, so when you used x11 socket with the `ecdata` directory where there was the regular file `.Xauthority`, docker failed to mount the directory `.Xauthority`. The solution may be deleting the file `.ecdata/.Xauthority` before you run the container with x11 socket.

The second problem: Was your username and password correct?

**Comment by hongyi-zhao** [comment]
> Try to `sudo rm ~/.ecdata/.Xauthority`?

Yep

> Did you run EasyConnect container in vnc with the same `ecdata` dir before the error happened?

Yep. I always use this same `ecdata` DIR  for all of my testings.

**Comment by hongyi-zhao** [comment]
> The second problem: Was your username and password correct?

To be frank, the day before yesterday, I have entered an incorrect password consecutively for several times accidentally and be blocked by the server. Since then, I have never been able to log in successfully.

**Comment by Hagb** [comment]
Is the problem solved?

**Comment by hongyi-zhao** [comment]
No, till now.

**Comment by Hagb** [comment]
Close the issue, for the `OCI runtime create failed` error has been solved. If there is any other problem, feel free to submit a new issue.

**Comment by hongyi-zhao** [comment]
I tried to build docker image the latest commit, but still encountered the same error:

![image](https://user-images.githubusercontent.com/11155854/108816128-8fcb5380-75f0-11eb-85c5-c2329501ee23.png)

**Comment by Hagb** [comment]
> I tried to build docker image the latest commit, but still failed with the following error:
> 
> ![image](https://user-images.githubusercontent.com/11155854/108816128-8fcb5380-75f0-11eb-85c5-c2329501ee23.png)

Can you login successfully in other platfarm?

**Comment by hongyi-zhao** [comment]
Tried on Android platform and meet the same error. Maybe the account has been disabled by the server.
